sky/dag.py

```diff
@@ -1,18 +1,16 @@
 """DAGs: user applications to be run."""
 import pprint
+import threading
 
 
-class DagContext:
-    """A global stack of Dags.
-
-    Currently, we only use one sky.Dag.
-    """
+class DagContext(threading.local):
```

Comment:
Just curious, will this work if it is not instantiated? Should we create a global variable in this file `dag_context = DAGContext()` and make the classmethod to be normal class member methods?

Reply:
Great catch! Pushed the changes, PTAL.

Re-ran smoke tests, all passed except:
- test_spot_storage: known access issue
- test_use_spot, test_spot: same issues as #1026,  keypairs in us-east-2